### PR TITLE
Add excludedocs option to the yum module

### PR DESCRIPTION
**Pull request by mrLarbi**
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

ansible/ansible-modules-core
Yum
##### ANSIBLE VERSION

```
ansible 2.1.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### SUMMARY

Add an option to exclude documentation files when installing à package with the Yum module (i.e the --excludedocs option from the rpm binary). Some rpm won't install without this option (e.g the perl-MIME-Base64-3.13-1.el6.rf.x86_64.rpm on CentOS 6.7). 

```
Before :

- name : install perl-MIME-Base64-3.13-1.el6.rf.x86_64.rpm
      yum:
        name=/tmp/perl-MIME-Base64-3.13-1.el6.rf.x86_64.rpm
        state=present

=> 

TASK [install perl-MIME-Base64-3.13-1.el6.rf.x86_64.rpm] ***********************
fatal: [centos]: FAILED! => {"changed": true, "failed": true, "msg": "\n\nTransaction Check Error:\n  file /usr/share/man/man3/MIME::Base64.3pm.gz from install of perl-MIME-Base64-3.13-1.el6.rf.x86_64 conflicts with file from package perl-4:5.10.1-141.el6_7.1.x86_64\n  file /usr/share/man/man3/MIME::QuotedPrint.3pm.gz from install of perl-MIME-Base64-3.13-1.el6.rf.x86_64 conflicts with file from package perl-4:5.10.1-141.el6_7.1.x86_64\n\nError Summary\n-------------\n\n", "rc": 1, "results": ["Loaded plugins: fastestmirror\nSetting up Install Process\nExamining /tmp/perl-MIME-Base64-3.13-1.el6.rf.x86_64.rpm: perl-MIME-Base64-3.13-1.el6.rf.x86_64\nMarking /tmp/perl-MIME-Base64-3.13-1.el6.rf.x86_64.rpm to be installed\nLoading mirror speeds from cached hostfile\n * base: centos.mirror.fr.planethoster.net\n * extras: mirror.ibcp.fr\n * updates: centos.mirror.ate.info\nResolving Dependencies\n--> Running transaction check\n---> Package perl-MIME-Base64.x86_64 0:3.13-1.el6.rf will be installed\n--> Finished Dependency Resolution\n\nDependencies Resolved\n\n================================================================================\n Package      Arch   Version       Repository                              Size\n================================================================================\nInstalling:\n perl-MIME-Base64\n              x86_64 3.13-1.el6.rf /perl-MIME-Base64-3.13-1.el6.rf.x86_64  96 k\n\nTransaction Summary\n================================================================================\nInstall       1 Package(s)\n\nTotal size: 96 k\nInstalled size: 96 k\nDownloading Packages:\nRunning rpm_check_debug\nRunning Transaction Test\n"]}

After :

    - name : install perl-MIME-Base64-3.13-1.el6.rf.x86_64.rpm
      yum:
        name=/tmp/perl-MIME-Base64-3.13-1.el6.rf.x86_64.rpm
        state=present
        excludedocs=yes

=>

TASK [install perl-MIME-Base64-3.13-1.el6.rf.x86_64.rpm] ***********************
changed: [centos]

```

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @mrLarbi for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

Core team: please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with 'needs_revision' or merge as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by robinro** [comment]
I don't think that messing with `/root/.rpmmacros` is a good idea. This feature should be implemented via arguement/env to yum. Messing with rpmmacros can interfer with other (simultaneous) installations. Also I'm not convinces that the solution works for general existing rpmmacros files.

**Comment by mrLarbi** [comment]
@robinro Hi robinro,

I also think that it would be much better to do it with an env variable or a command line option. The problem is that rpm doesn't seem to use env variables to define macros, and the yum command doesn't have an option to handle that either (the only way to do it is by using the yum.conf).

**Comment by mrLarbi** [comment]
@robinro I'v found the yum-tsflags package, which allows to specify the nodocs option to the yum command line. It needs to be installed, but it is indeed better than using the rpmmacros file. I'm going to change my implementation. :)

File: packaging/os/yum.py [hunk 1]
```diff
@@ -677,6 +683,13 @@ def install(module, items, repoq, yum_basecmd, conf_file, en_repos, dis_repos):
         pkgs.append(pkg)
 
     if pkgs:
+        
+        if(module.params['excludedocs']):
+            # Needed for the --tsflags nodocs option.
+            module.run_command(['yum', 'install', "yum-tsflags", '-y'], check_rc=True)
```
Comment (robinro) [inline packaging/os/yum.py]: Don't install the yum-tsflags package here. Instead put it in `requirements` and mention it in the documentation.
Then people can choose whether they want to use this feature and flag or not.

**Review by robinro — COMMENTED** [review]
Looks a lot better now :smile:  Didn't run it though.

File: packaging/os/yum.py (hunk 1 above)
Comment (mrLarbi) [reply packaging/os/yum.py]: Done ! :)

**Pull request closed without merging**